solidity/contracts/ETHStaking.sol

```diff
@@ -0,0 +1,66 @@
+/**
+▓▓▌ ▓▓ ▐▓▓ ▓▓▓▓▓▓▓▓▓▓▌▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▄
+▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▌▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓
+  ▓▓▓▓▓▓    ▓▓▓▓▓▓▓▀    ▐▓▓▓▓▓▓    ▐▓▓▓▓▓   ▓▓▓▓▓▓     ▓▓▓▓▓   ▐▓▓▓▓▓▌   ▐▓▓▓▓▓▓
+  ▓▓▓▓▓▓▄▄▓▓▓▓▓▓▓▀      ▐▓▓▓▓▓▓▄▄▄▄         ▓▓▓▓▓▓▄▄▄▄         ▐▓▓▓▓▓▌   ▐▓▓▓▓▓▓
+  ▓▓▓▓▓▓▓▓▓▓▓▓▓▀        ▐▓▓▓▓▓▓▓▓▓▓         ▓▓▓▓▓▓▓▓▓▓▌        ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓
+  ▓▓▓▓▓▓▀▀▓▓▓▓▓▓▄       ▐▓▓▓▓▓▓▀▀▀▀         ▓▓▓▓▓▓▀▀▀▀         ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▀
+  ▓▓▓▓▓▓   ▀▓▓▓▓▓▓▄     ▐▓▓▓▓▓▓     ▓▓▓▓▓   ▓▓▓▓▓▓     ▓▓▓▓▓   ▐▓▓▓▓▓▌
+▓▓▓▓▓▓▓▓▓▓ █▓▓▓▓▓▓▓▓▓ ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓  ▓▓▓▓▓▓▓▓▓▓
+▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓ ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓  ▓▓▓▓▓▓▓▓▓▓
+
+                           Trust math, not hardware.
+*/
+
+pragma solidity 0.5.17;
+
+import "@keep-network/keep-core/contracts/KeepRegistry.sol";
+import "@keep-network/keep-core/contracts/Authorizations.sol";
+import "@keep-network/keep-core/contracts/StakeDelegatable.sol";
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+/// @title ETH Staking
+/// @notice A staking contract for ETH staking. An owner of the ETH can delegate
+/// ETH as a stake to an operator. The value of ETH the owner is willing to stake
+/// should be deposited in `ETHBonding` contract for the given operator.
```

Comment:
I'm not sure about how to name this contract, I had ideas like `ETHStaking` and `ETHDelegating`. Decided on `ETHStaking` to be consistent with `TokenStaking`. 
It is not strictly about staking as in the token staking, but on the other hand, we are naming the functionality ETH-only staking, which means we stake ETH. The bond will become a stake.

We could combine ETHBonding and ETHStaking contracts but the resulting contract may get too big, too complex and too confusing at some point.

It's important for us to decide on the naming now as this will impact other places where we want to use this contract.

Waiting for your thoughts and suggestions.

Reply:
I'd probably go with `EthDelegation`.

`TokenStaking` is about KEEP token staking and we are staking neither KEEP nor ETH in `EthStaking`. What we do instead, is that we are defining *a delegation* of ETH - from owner to operator, using the given beneficiary and authorizer.

Reply:
Thoughts on `EthDelegating` or `EthDelegations`? To me `EthDelegation`suggests a single delegation unit, something appropriate for a structure or an event.

`EthDelegating` seems to be consistent with `TokenStaking`, `KeepBonding` and `EthBonding`.

Reply:
I ❤️ `EthDelegating`, it's consistent with all the other contract names you mentioned.

Reply:
5c0c536cb1f9fb98eeeb744d93328687760c993c